compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -5928,6 +5935,16 @@ private Object parseLong(Node literal, String originalNodeValue,
     }
 
     private String getHexNodeValue(String value) {
+        // Remove exponent indicator if it is not followed by a digit
+        if (value.matches(".*[pP][+-]?")) {
+            value = value.replaceAll("[pP][+-]?", "");
+        }
+        
+        // Add 0 if there is no hex digit precedes hex indicator
+        if (value.matches("0[xX][pP].*")) {
+            value = value.replaceAll("[pP]", "0p");
+        }
```

Comment:
Here also the same thing,
Example 1
```ballerina
public function main() {
    0f a = 0Xp34; // type set as 0f
}
```
Gives error as
```
ERROR [test3.bal:(22:12,22:17)] missing hex number after hex indicator
```
Example 2
```ballerina
public function main() {
    10f a = 0Xp34; // identify type as float
}
```
Gives error as
```
ERROR [test3.bal:(22:13,22:18)] incompatible types: expected '10ff', found 'float'
ERROR [test3.bal:(22:13,22:18)] missing hex number after hex indicator
```

Reply:
Fixed in [932a51e](https://github.com/ballerina-platform/ballerina-lang/pull/35646/commits/932a51ecb0f19f3946d3394d71c2ed14ad5dcb73)